### PR TITLE
Testing strawman

### PR DESCRIPTION
**Pull request by dfreeman**
Just playing around with what testing for the HTMLBars plugin could look like. Love to get your thoughts whenever you have a little time to take a look.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
   "scripts": {
     "build": "ember build",
     "start": "ember server",
-    "test": "ember try:each"
+    "test": "npm run test-lib && npm run test-ember",
+    "test-ember": "ember try:each",
+    "test-lib": "tape tests-node/**/*-test.js"
```
Comment (dfreeman) [inline package.json:10]: Don't have strong opinions on the directory structure and framework in use — this was just quick to get up and running. Might have been nice to put these in a subdirectory of `tests`, but I couldn't immediately come up with a way to filter those out from attempting to also run in the browser.

File: tests-node/htmlbars-plugin-test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  input: 'local-class="foo"',
+  statementOutput: 'class=(concat styles.foo)', // FIXME superfluous concat
+  elementOutput: 'class="{{styles.foo}}"'
+});
+
+testTransformation('appending to a class attribute', {
+  input: 'class="x" local-class="foo"',
+  statementOutput: 'class=(concat "x" " " styles.foo)',
+  elementOutput: 'class="x {{styles.foo}}"'
+});
```
Comment (dfreeman) [inline tests-node/htmlbars-plugin-test.js:15]: I could see there being a need for tests with finer-grained assertions than this approach allows, but this kind of setup as a starting point should make it easy to cover a lot of cases quickly, I think ¯\_(ツ)_/¯

**Comment by jasonmit** [comment]
Looks good on the surface and really appreciate you taking the testing story on 👍

File: tests-node/htmlbars-plugin-test.js (hunk 2 above)
Comment (jasonmit) [inline tests-node/htmlbars-plugin-test.js:15]: It should be fine, for the more specific tests we can create other helpers or just be verbose/explicit in those tests

File: package.json (hunk 1 above)
Comment (jasonmit) [inline package.json:10]: I haven't used tape before, mainly mocha.  But I'm fine with using it.

As for tests-node, that's exactly where I've put them in the past.

**Pull request merged**